### ISSUE TITLE
[WS3-D] LORA_PRESETS config + UI training panel preset dropdown

### ISSUE THREAD

**Issue by mgcronin** [openzigs/openzigs#933]
﻿## Description

Add a `LORA_PRESETS` config (in `config/default.json` or new `config/lora-presets.json`) capturing best-practice training params per base model. Surface a preset dropdown in the UI training panel with an "Advanced" toggle to override individual fields.

**Parent Epic**: #924
**Workstream**: WS3 (LoRA Training)
**Story Points**: 3

## Context

Right now users have to know the right values for rank, alpha, lr, steps, batch size, gradient accumulation, mixed precision, and resolution per model architecture. We have published best-practice values from the diffusers/PEFT community â€” bake them in.

## Acceptance Criteria

- [ ] New config file `config/lora-presets.json` (preferred, isolated) OR new key `loraPresets` in `config/default.json`
- [ ] Schema:
  ```json
  {
    "sdxl":         { "rank": 16, "alpha": 32, "lr": 1e-4, "steps": 750, "batch": 1, "grad_accum": 1, "mixed": "fp16", "resolution": 1024 },
    "flux-dev":     { "rank": 16, "alpha": 32, "lr": 1e-4, "steps": 750, "batch": 1, "grad_accum": 4, "mixed": "bf16", "resolution": 512 },
    "flux-schnell": { "rank": 16, "alpha": 32, "lr": 1e-4, "steps": 500, "batch": 1, "grad_accum": 4, "mixed": "bf16", "resolution": 512 },
    "sd15":         { "rank": 8,  "alpha": 16, "lr": 1e-4, "steps": 500, "batch": 1, "grad_accum": 1, "mixed": "fp16", "resolution": 512 }
  }
  ```
- [ ] Server endpoint `GET /api/training/presets` returns the merged config (default + user overrides)
- [ ] User can override presets in `~/.openzigs/config.json` (existing config-layering pattern)
- [ ] UI training panel:
  - Preset dropdown (defaults based on selected base_model from WS3-B)
  - "Advanced" disclosure expands to show all fields for editing
  - Switching preset updates all fields; manual edits mark the form as "Custom (modified from {preset})"
- [ ] Server applies preset values when payload omits them; explicit payload values always win
- [ ] Vitest tests for UI: preset selection, override behavior, "Custom" indicator
- [ ] Vitest test for server merge logic
- [ ] `docs/USER_GUIDE.md` documents the presets and how to override
- [ ] `CHANGELOG.md` updated
- [ ] Quality gate: `pnpm lint && pnpm typecheck && pnpm test && cd ui && npx next build`

### Files to Create
- `config/lora-presets.json`

### Files to Modify
- `src/config/index.ts` â€” load `lora-presets.json` with the existing config-layering pattern
- `src/api/admin.ts` (or new router) â€” `GET /api/training/presets`
- `sidecars/image-gen/server_cuda.py` â€” accept presets via payload (server already passes through)
- `ui/components/.../training-panel.tsx` â€” preset dropdown + advanced toggle
- `ui/lib/api.ts` â€” typed client for presets
- `docs/USER_GUIDE.md`, `CHANGELOG.md`

## Dependencies

- **Blocked by**: WS3-A (alpha = 2*rank default â€” presets assume this), WS3-B (base_model dropdown drives preset selection)
- **Blocks**: none

## Security Considerations

- Validate preset overrides server-side (numeric ranges: rank 1-128, alpha 1-256, lr 1e-6 to 1e-2, etc.)
- Reject preset payloads with unknown fields (avoid passthrough of arbitrary kwargs to trainer subprocess)
- User config file at `~/.openzigs/config.json` retains `0o600` perms (existing pattern)

## Definition of Done

- [ ] All acceptance criteria checked
- [ ] PR opened with `Closes #THIS_ISSUE`
- [ ] CI green
- [ ] Manual smoke: open training panel, select FLUX â†’ fields populate with FLUX preset; switch to SDXL â†’ fields update
